inc/stm32_compat.h

```diff
@@ -0,0 +1,59 @@
+#ifndef _STM32_COMPAT_H_
+#define _STM32_COMPAT_H_
+
+#ifndef PLATFORMIO
+
+#include <stm32.h>
+
+#else // PLATFORMIO
+/* modify bitfield */
+#define _BMD(reg, msk, val)     (reg) = (((reg) & ~(msk)) | (val))
+/* set bitfield */
+#define _BST(reg, bits)         (reg) = ((reg) | (bits))
+/* clear bitfield */
+#define _BCL(reg, bits)         (reg) = ((reg) & ~(bits))
+/* wait until bitfield set */
+#define _WBS(reg, bits)         while(((reg) & (bits)) == 0)
+/* wait until bitfield clear */
+#define _WBC(reg, bits)         while(((reg) & (bits)) != 0)
+/* wait for bitfield value */
+#define _WVL(reg, msk, val)     while(((reg) & (msk)) != (val))
+/* bit value */
+#define _BV(bit)                (0x01 << (bit))
+
+#if defined(STM32F0xx)
+    #include <stm32f0xx.h>
+#elif defined(STM32F1xx)
+    #include <stm32f1xx.h>
+#elif defined(STM32F2xx)
+    #include <stm32f2xx.h>
+#elif defined(STM32F3xx)
+    #include <stm32f3xx.h>
+#elif defined(STM32F4xx)
+    #include <stm32f4xx.h>
+#elif defined(STM32F7xx)
+    #include <stm32f7xx.h>
+#elif defined(STM32H7xx)
+    #include <stm32h7xx.h>
+#elif defined(STM32L0xx)
+    #include <stm32l0xx.h>
+#elif defined(STM32L1xx)
+    #include <stm32l1xx.h>
+#elif defined(STM32L4xx)
+    #include <stm32l4xx.h>
+#elif defined(STM32L5xx)
+    #include <stm32l5xx.h>
+#elif defined(STM32G0xx)
+    #include <stm32g0xx.h>
+#elif defined(STM32G4xx)
+    #include <stm32g4xx.h>
+#elif defined(STM32WBxx)
```

Comment:
What's the rationale behind changing this logic so that it looks at these long `xx` macros?

The original stm32.h observes `STM32L0` instead of `STM32L0xx`. After this change I need to have `build_flags = -DSTM32L0xx` in order to pass the build.

Reply:
these definitions are added automatically by pio once the user selects the mcu, and (contrary to makefile system) this is not done manually.

Reply:
Experimenting on `board = board = disco_l072cz_lrwan1
`. I think there are weird macro differences between frameworks... perhaps also `boards`. Here listing only `STM32` macro defs:

libopencm3: `-DSTM32L072xx -DSTM32L0`
cmsis: `-DSTM32L072xx`
stm32cube: `-DSTM32L072xx`
arduino (builds): `-DSTM32L072xx -DSTM32L0xx`

Quite pathetic I guess. I thought DSTM32L0 would be shared among all the cases.

Reply:
We might want to expand the scope of what that file does and call it `stm32_sanity.h`

Reply:
Or rather go the straightforward way and use [`extraScript` ](https://docs.platformio.org/en/latest/manifests/library-json/fields/build/extrascript.html) to:
- Add proper `extra_flags = -DSTM??`
- Use explicit `src_filter` to only build the relevant sources

`stm32_compat.h` would not require the PIO specialty.

Reply:
Then unlike other libraries, it wouldn't Just Work, making it require manual user intervention, which is less than ideal.

Reply:
Since most other frameworks use the `...xx` convension, I propose to change `stm32h` to use those instead.

Reply:
I don't see a reason for additional end user workload, but I agree that tuning the ifdefs should be good enough.